Lib/test/test_enum.py

```diff
@@ -2036,6 +2036,35 @@ class GoodStrEnum(StrEnum):
             two = '2'
             three = b'3', 'ascii'
             four = b'4', 'latin1', 'strict'
+        self.assertEqual(GoodStrEnum.one, '1')
+        self.assertEqual(GoodStrEnum.one, str(GoodStrEnum.one))
```

Comment:
Is it what you meant? It does not test that `str(GoodStrEnum.one)` is equal to '1', it tests that it is equal to `GoodStrEnum.one` which is a different thing. `GoodStrEnum.one` may be equal to several different strings. Maybe you meant `self.assertEqual(str(GoodStrEnum.one), '1')`?

I would add also test for `format(GoodStrEnum.one)`.

Reply:
It's a two-step process:

- check that `GoodStrEnum.one` is the string `"1"`
- check that `str(GoodStrEnum.one)` is also the string `"1"`

Adding `format()` test.

Reply:
Format test added.

Reply:
But from `a == b` and `a == c` it does not follow that `b == c`. Why do not test that `str(GoodStrEnum.one)` equals to string `"1"` directly?

Reply:
Okay, direct test added.  I kept the other test as well, since `b == c` should be true in this case.